compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -103,18 +103,14 @@ fn reachable_non_generics_provider(tcx: TyCtxt<'_>, cnum: CrateNum) -> DefIdMap<
             }
         })
         .map(|def_id| {
-            let (export_level, used) = if special_runtime_crate {
-                let name = tcx.symbol_name(Instance::mono(tcx, def_id.to_def_id())).name;
-                // We won't link right if these symbols are stripped during LTO.
-                let used = match name {
-                    "rust_eh_personality"
-                    | "rust_eh_register_frames"
-                    | "rust_eh_unregister_frames" => true,
-                    _ => false,
-                };
-                (SymbolExportLevel::Rust, used)
+            // We won't link right if this symbol is stripped during LTO.
+            let name = tcx.symbol_name(Instance::mono(tcx, def_id.to_def_id())).name;
+            let used = name == "rust_eh_personality";
+
+            let export_level = if special_runtime_crate {
```

Comment:
This means that `rust_eh_personality` is now exported from cdylibs as it doesn't use `SymbolExportLevel::Rust`, right? That is going to give troubles when depending on two cdylibs compiled by different versions of rustc if we change the eh_personality implementation as one version of this symbol will override the other.

Reply:
I think `symbol_export_level` still returns `Rust` for `rust_eh_personality` since it doesn't have any attribute such as `no_mangle` that marks it as externally visible.

Reply:
I just checked it and you are indeed right. `rust_eh_personality` is not exported.